Refactoring: rename FullscreenManager to DocumentFullscreen

nt1m opened the pull request:
#### be9370a1451a1af3a6a380ffc37e1081c34cdc53
<pre>
Refactoring: rename FullscreenManager to DocumentFullscreen
<a href="https://bugs.webkit.org/show_bug.cgi?id=259978">https://bugs.webkit.org/show_bug.cgi?id=259978</a>
<a href="https://rdar.apple.com/113633238">rdar://113633238</a>

Reviewed by Jer Noble.

FullscreenManager is an implementation of a IDL extension called Document+Fullscreen.
We already have a DocumentFullscreen.cpp/h file which calls out to FullscreenManager.
And so the name &quot;FullscreenManager&quot; is not strictly correct. It doesn&apos;t &quot;manage&quot; fullscreen;
it&apos;s merely a way to collect Document&apos;s fullscreen implementation into a separate class.

Rename FullscreenManager to DocumentFullscreen and move its implementation into the existing
DocumentFullscreen.cpp/h files. Rename Document::fullscreenManager() to just Document::fullscreen().

No functional changes here, just renames, so covered by existing tests.

* LayoutTests/media/media-fullscreen.js:
(async beginfullscreen):
* Source/WebCore/Headers.cmake:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/SelectorCheckerTestFunctions.h:
(WebCore::matchesAnimatingFullscreenTransitionPseudoClass):
(WebCore::matchesFullscreenDocumentPseudoClass):
(WebCore::matchesInWindowFullscreenPseudoClass):
* Source/WebCore/dom/Document+Fullscreen.idl:
* Source/WebCore/dom/Document.cpp:
(WebCore::m_syncData):
(WebCore::Document::commonTeardown):
(WebCore::Document::ensureFullscreen):
(WebCore::Document::fullscreen):
(WebCore::Document::fullscreen const):
(WebCore::Document::checkedFullscreen):
(WebCore::Document::checkedFullscreen const):
(WebCore::eventTargetElementForDocument):
(WebCore::Document::ensureFullscreenManager): Deleted.
(WebCore::Document::fullscreenManager): Deleted.
(WebCore::Document::fullscreenManager const): Deleted.
(WebCore::Document::checkedFullscreenManager): Deleted.
(WebCore::Document::checkedFullscreenManager const): Deleted.
* Source/WebCore/dom/Document.h:
(WebCore::Document::fullscreenIfExists):
(WebCore::Document::fullscreenIfExists const):
(WebCore::Document::fullscreenManagerIfExists): Deleted.
(WebCore::Document::fullscreenManagerIfExists const): Deleted.
* Source/WebCore/dom/DocumentFullscreen.cpp:
(WebCore::DocumentFullscreen::DocumentFullscreen):
(WebCore::DocumentFullscreen::fullscreenEnabled):
(WebCore::DocumentFullscreen::isFullscreenEnabled const):
(WebCore::DocumentFullscreen::fullscreenElement const):
(WebCore::DocumentFullscreen::requestFullscreenForElement):
(WebCore::DocumentFullscreen::willEnterFullscreen):
(WebCore::DocumentFullscreen::elementEnterFullscreen):
(WebCore::DocumentFullscreen::didEnterFullscreen):
(WebCore::DocumentFullscreen::isSimpleFullscreenDocument const):
(WebCore::documentsToUnfullscreen):
(WebCore::clearFullscreenFlags):
(WebCore::DocumentFullscreen::exitFullscreen):
(WebCore::DocumentFullscreen::webkitExitFullscreen):
(WebCore::DocumentFullscreen::finishExitFullscreen):
(WebCore::DocumentFullscreen::willExitFullscreen):
(WebCore::DocumentFullscreen::didExitFullscreen):
(WebCore::DocumentFullscreen::exitRemovedFullscreenElement):
(WebCore::DocumentFullscreen::fullyExitFullscreen):
(WebCore::DocumentFullscreen::dispatchPendingEvents):
(WebCore::DocumentFullscreen::queueFullscreenChangeEventForDocument):
(WebCore::DocumentFullscreen::isAnimatingFullscreen const):
(WebCore::DocumentFullscreen::setAnimatingFullscreen):
(WebCore::DocumentFullscreen::logChannel const):
(WebCore::DocumentFullscreen::webkitFullscreenEnabled): Deleted.
(WebCore::DocumentFullscreen::webkitFullscreenElement): Deleted.
(WebCore::DocumentFullscreen::webkitIsFullScreen): Deleted.
(WebCore::DocumentFullscreen::webkitFullScreenKeyboardInputAllowed): Deleted.
(WebCore::DocumentFullscreen::webkitCurrentFullScreenElement): Deleted.
(WebCore::DocumentFullscreen::webkitCancelFullScreen): Deleted.
* Source/WebCore/dom/DocumentFullscreen.h:
* Source/WebCore/dom/DocumentOrShadowRootFullscreen.cpp:
(WebCore::DocumentOrShadowRootFullscreen::fullscreenElement):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::removedFromAncestor):
(WebCore::Element::requestFullscreen):
* Source/WebCore/dom/EventPath.cpp:
* Source/WebCore/dom/FullscreenManager.cpp: Removed.
* Source/WebCore/dom/FullscreenManager.h: Removed.
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::elementIsHidden const):
(WebCore::HTMLMediaElement::isFullscreen const):
(WebCore::HTMLMediaElement::isStandardFullscreen const):
(WebCore::HTMLMediaElement::enterFullscreen):
(WebCore::HTMLMediaElement::exitFullscreen):
* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::MediaElementSession::canShowControlsManager const):
(WebCore::MediaElementSession::updateMediaUsageIfChanged):
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::isKeyEventAllowedInFullScreen const):
(WebCore::EventHandler::internalKeyEvent):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::documentBackgroundColor const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::updateRendering):
(WebCore::Page::outermostFullscreenDocument const):
* Source/WebCore/page/ScreenOrientation.cpp:
(WebCore::ScreenOrientation::lock):
* Source/WebCore/page/ios/ContentChangeObserver.cpp:
(WebCore::isHiddenBehindFullscreenElement):
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::RenderLayerBacking):
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::isDescendantOfFullScreenLayer):
* Source/WebCore/rendering/RenderVideo.cpp:
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjust const):
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):
* Source/WebCore/style/StyleSharingResolver.cpp:
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebCore/testing/Internals.cpp:
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
(WebKit::WebFullScreenManager::willEnterFullScreen):
(WebKit::WebFullScreenManager::willExitFullScreen):
(WebKit::collectFullscreenElementsFromElement):
(WebKit::WebFullScreenManager::setAnimatingFullScreen):
(WebKit::WebFullScreenManager::requestRestoreFullScreen):
(WebKit::WebFullScreenManager::requestExitFullScreen):
(WebKit::WebFullScreenManager::handleEvent):
(WebKit::WebFullScreenManager::mainVideoElementTextRecognitionTimerFired):
(WebKit::WebFullScreenManager::enterFullScreenForOwnerElements):
(WebKit::WebFullScreenManager::exitFullScreenInMainFrame):
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMDocumentGtk.cpp:
(webkit_dom_document_webkit_cancel_fullscreen):
(webkit_dom_document_webkit_exit_fullscreen):
(webkit_dom_document_get_webkit_is_fullscreen):
(webkit_dom_document_get_webkit_fullscreen_keyboard_input_allowed):
(webkit_dom_document_get_webkit_current_fullscreen_element):
(webkit_dom_document_get_webkit_fullscreen_enabled):
(webkit_dom_document_get_webkit_fullscreen_element):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::createWindow):
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::dispatchDidStartProvisionalLoad):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::elementDidFocus):
* Source/WebKitLegacy/mac/DOM/DOMDocument.mm:
(-[DOMDocument webkitCurrentFullScreenElement]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.mm:
(WebChromeClient::createWindow):
* Source/WebKitLegacy/mac/WebCoreSupport/WebKitFullScreenListener.mm:
(-[WebKitFullScreenListener webkitWillEnterFullScreen]):
(-[WebKitFullScreenListener webkitWillExitFullScreen]):
* Source/WebKitLegacy/mac/WebView/WebFullScreenController.mm:
(-[WebFullScreenController requestExitFullScreen]):
(-[WebFullScreenController _manager]):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _didStartProvisionalLoadForFrame:]):

Canonical link: <a href="https://commits.webkit.org/291091@main">https://commits.webkit.org/291091@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a0af2a04110baf5a2e178f877ef66513e125ba3f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91936 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/11475 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/1011 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/96872 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/42542 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/93986 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/11815 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/19949 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/70549 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28030 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/94937 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/9009 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/83272 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50876 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/8739 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/867 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/41757 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/79061 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/867 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/98901 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/19059 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/14077 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/79580 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/19311 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/79118 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/78806 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/23342 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/648 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/12097 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/14603 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/19040 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/24249 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/18737 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/22196 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/20488 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/WebCore/css/SelectorCheckerTestFunctions.h:
```diff
@@ -411,7 +411,7 @@ ALWAYS_INLINE bool matchesFullscreenPseudoClass(const Element& element)
 ALWAYS_INLINE bool matchesAnimatingFullscreenTransitionPseudoClass(const Element& element)
 {
-    CheckedPtr fullscreenManager = element.document().fullscreenManagerIfExists();
+    CheckedPtr fullscreenManager = element.document().fullscreenIfExists();
```
Shouldn't this be renamed too (and all that follows)?

jyavenard reviewed (commented):
Maybe it's because I'm too used to it.

But having things like `fullscreen().fullyExitFullscreen()` really expose how redundant the naming of some methods are.

jyavenard commented on Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
```diff
@@ -37,8 +37,8 @@
 #include "WebPage.h"
```
What about WebFullScreenManager ? that seems out of place too

nt1m replied:
I think this is fine as `WebFullScreenManager`, we could perhaps make that `FullscreenManager`, but there's little benefit to renaming it.

jyavenard commented on Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
```diff
@@ -565,7 +565,7 @@ void WebFullScreenManager::requestRestoreFullScreen(CompletionHandler<void(bool)
     ALWAYS_LOG(LOGIDENTIFIER, "<", element->tagName(), " id=\"", element->getIdAttribute(), "\">");
     WebCore::UserGestureIndicator gestureIndicator(WebCore::IsProcessingUserGesture::Yes, &element->document());
-    element->document().fullscreenManager().requestFullscreenForElement(*element, WebCore::FullscreenManager::ExemptIFrameAllowFullscreenRequirement, [completionHandler = WTFMove(completionHandler)] (auto result) mutable {
+    element->document().fullscreen().requestFullscreenForElement(*element, WebCore::DocumentFullscreen::ExemptIFrameAllowFullscreenRequirement, [completionHandler = WTFMove(completionHandler)] (auto result) mutable {
```
is requestFullscreenForElement really required?

rather than `element->document().fullscreen().requestFullscreen(....)`

nt1m replied:
Yeah we could drop `forElement` potentially.

nt1m committed:
Refactoring: rename FullscreenManager to DocumentFullscreen
https://bugs.webkit.org/show_bug.cgi?id=259978
rdar://113633238

Reviewed by Jer Noble.

FullscreenManager is an implementation of a IDL extension called Document+Fullscreen.
We already have a DocumentFullscreen.cpp/h file which calls out to FullscreenManager.
And so the name "FullscreenManager" is not strictly correct. It doesn't "manage" fullscreen;
it's merely a way to collect Document's fullscreen implementation into a separate class.

Rename FullscreenManager to DocumentFullscreen and move its implementation into the existing
DocumentFullscreen.cpp/h files. Rename Document::fullscreenManager() to just Document::fullscreen().

No functional changes here, just renames, so covered by existing tests.

* LayoutTests/media/media-fullscreen.js:
(async beginfullscreen):
* Source/WebCore/Headers.cmake:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/SelectorCheckerTestFunctions.h:
(WebCore::matchesAnimatingFullscreenTransitionPseudoClass):
(WebCore::matchesFullscreenDocumentPseudoClass):
(WebCore::matchesInWindowFullscreenPseudoClass):
* Source/WebCore/dom/Document+Fullscreen.idl:
* Source/WebCore/dom/Document.cpp:
(WebCore::m_syncData):
(WebCore::Document::commonTeardown):
(WebCore::Document::ensureFullscreen):
(WebCore::Document::fullscreen):
(WebCore::Document::fullscreen const):
(WebCore::Document::checkedFullscreen):
(WebCore::Document::checkedFullscreen const):
(WebCore::eventTargetElementForDocument):
(WebCore::Document::ensureFullscreenManager): Deleted.
(WebCore::Document::fullscreenManager): Deleted.
(WebCore::Document::fullscreenManager const): Deleted.
(WebCore::Document::checkedFullscreenManager): Deleted.
(WebCore::Document::checkedFullscreenManager const): Deleted.
* Source/WebCore/dom/Document.h:
(WebCore::Document::fullscreenIfExists):
(WebCore::Document::fullscreenIfExists const):
(WebCore::Document::fullscreenManagerIfExists): Deleted.
(WebCore::Document::fullscreenManagerIfExists const): Deleted.
* Source/WebCore/dom/DocumentFullscreen.cpp:
(WebCore::DocumentFullscreen::DocumentFullscreen):
(WebCore::DocumentFullscreen::fullscreenEnabled):
(WebCore::DocumentFullscreen::isFullscreenEnabled const):
(WebCore::DocumentFullscreen::fullscreenElement const):
(WebCore::DocumentFullscreen::requestFullscreenForElement):
(WebCore::DocumentFullscreen::willEnterFullscreen):
(WebCore::DocumentFullscreen::elementEnterFullscreen):
(WebCore::DocumentFullscreen::didEnterFullscreen):
(WebCore::DocumentFullscreen::isSimpleFullscreenDocument const):
(WebCore::documentsToUnfullscreen):
(WebCore::clearFullscreenFlags):
(WebCore::DocumentFullscreen::exitFullscreen):
(WebCore::DocumentFullscreen::webkitExitFullscreen):
(WebCore::DocumentFullscreen::finishExitFullscreen):
(WebCore::DocumentFullscreen::willExitFullscreen):
(WebCore::DocumentFullscreen::didExitFullscreen):
(WebCore::DocumentFullscreen::exitRemovedFullscreenElement):
(WebCore::DocumentFullscreen::fullyExitFullscreen):
(WebCore::DocumentFullscreen::dispatchPendingEvents):
(WebCore::DocumentFullscreen::queueFullscreenChangeEventForDocument):
(WebCore::DocumentFullscreen::isAnimatingFullscreen const):
(WebCore::DocumentFullscreen::setAnimatingFullscreen):
(WebCore::DocumentFullscreen::logChannel const):
(WebCore::DocumentFullscreen::webkitFullscreenEnabled): Deleted.
(WebCore::DocumentFullscreen::webkitFullscreenElement): Deleted.
(WebCore::DocumentFullscreen::webkitIsFullScreen): Deleted.
(WebCore::DocumentFullscreen::webkitFullScreenKeyboardInputAllowed): Deleted.
(WebCore::DocumentFullscreen::webkitCurrentFullScreenElement): Deleted.
(WebCore::DocumentFullscreen::webkitCancelFullScreen): Deleted.
* Source/WebCore/dom/DocumentFullscreen.h:
* Source/WebCore/dom/DocumentOrShadowRootFullscreen.cpp:
(WebCore::DocumentOrShadowRootFullscreen::fullscreenElement):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::removedFromAncestor):
(WebCore::Element::requestFullscreen):
* Source/WebCore/dom/EventPath.cpp:
* Source/WebCore/dom/FullscreenManager.cpp: Removed.
* Source/WebCore/dom/FullscreenManager.h: Removed.
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::elementIsHidden const):
(WebCore::HTMLMediaElement::isFullscreen const):
(WebCore::HTMLMediaElement::isStandardFullscreen const):
(WebCore::HTMLMediaElement::enterFullscreen):
(WebCore::HTMLMediaElement::exitFullscreen):
* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::MediaElementSession::canShowControlsManager const):
(WebCore::MediaElementSession::updateMediaUsageIfChanged):
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::isKeyEventAllowedInFullScreen const):
(WebCore::EventHandler::internalKeyEvent):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::documentBackgroundColor const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::updateRendering):
(WebCore::Page::outermostFullscreenDocument const):
* Source/WebCore/page/ScreenOrientation.cpp:
(WebCore::ScreenOrientation::lock):
* Source/WebCore/page/ios/ContentChangeObserver.cpp:
(WebCore::isHiddenBehindFullscreenElement):
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::RenderLayerBacking):
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::isDescendantOfFullScreenLayer):
* Source/WebCore/rendering/RenderVideo.cpp:
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjust const):
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):
* Source/WebCore/style/StyleSharingResolver.cpp:
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebCore/testing/Internals.cpp:
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
(WebKit::WebFullScreenManager::willEnterFullScreen):
(WebKit::WebFullScreenManager::willExitFullScreen):
(WebKit::collectFullscreenElementsFromElement):
(WebKit::WebFullScreenManager::setAnimatingFullScreen):
(WebKit::WebFullScreenManager::requestRestoreFullScreen):
(WebKit::WebFullScreenManager::requestExitFullScreen):
(WebKit::WebFullScreenManager::handleEvent):
(WebKit::WebFullScreenManager::mainVideoElementTextRecognitionTimerFired):
(WebKit::WebFullScreenManager::enterFullScreenForOwnerElements):
(WebKit::WebFullScreenManager::exitFullScreenInMainFrame):
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMDocumentGtk.cpp:
(webkit_dom_document_webkit_cancel_fullscreen):
(webkit_dom_document_webkit_exit_fullscreen):
(webkit_dom_document_get_webkit_is_fullscreen):
(webkit_dom_document_get_webkit_fullscreen_keyboard_input_allowed):
(webkit_dom_document_get_webkit_current_fullscreen_element):
(webkit_dom_document_get_webkit_fullscreen_enabled):
(webkit_dom_document_get_webkit_fullscreen_element):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::createWindow):
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::dispatchDidStartProvisionalLoad):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::elementDidFocus):
* Source/WebKitLegacy/mac/DOM/DOMDocument.mm:
(-[DOMDocument webkitCurrentFullScreenElement]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.mm:
(WebChromeClient::createWindow):
* Source/WebKitLegacy/mac/WebCoreSupport/WebKitFullScreenListener.mm:
(-[WebKitFullScreenListener webkitWillEnterFullScreen]):
(-[WebKitFullScreenListener webkitWillExitFullScreen]):
* Source/WebKitLegacy/mac/WebView/WebFullScreenController.mm:
(-[WebFullScreenController requestExitFullScreen]):
(-[WebFullScreenController _manager]):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _didStartProvisionalLoadForFrame:]):

Canonical link: https://commits.webkit.org/291091@main
